app/models/template_division.rb

```diff
@@ -46,4 +31,30 @@ def destroy_associated_objects
     criterion.destroy
   end
 
+
+  def set_defaults_for_associated_criteria_assignment_files_join
+    if criteria_assignment_files_join.nil?
+      assignment_file = AssignmentFile.create(
+        filename: "#{exam_template.name}-#{label}.pdf",
+        assignment_id: exam_template.assignment.id
+      )
+      criterion = FlexibleCriterion.create(
+        max_mark: 1.0,
+        name: "#{exam_template.name}-#{label}",
```

Comment:
This isn't what we want. The criterion should only have the `label` as its name; this will allow different exam templates to use the same criteria. Here you can use `find_or_initialize_by` with an assignment id; if it's new, set the max_mark to 1.